app/models/ext_management_system.rb

```diff
@@ -426,6 +426,24 @@ def enable!
     update!(:enabled => true)
   end
 
+  # Wait until all associated workers are dead to destroy this ems
+  def orchestrate_destroy
+    disable_ems if enabled?
+
+    workers = MiqWorker.find_current_or_starting.where(:queue_name => "ems_#{id}") #better way to get 'queue_name'?
+    if workers.count == 0
```

Comment:
maybe these checks can be put into a rails `before_destroy` callback and cancel the destroy via normal rails methods?